code_generator/GdkCodeGenerator/Templates/UnityGameObjectComponentDispatcherGenerator.tt

```diff
@@ -60,21 +62,35 @@ namespace <#= qualifiedNamespace #>
                 return <#= unityComponentDefinition.Id #>;
             }
 
-            public override void InvokeOnComponentUpdateUserCallbacks(GameObjectDispatcherSystem gameObjectDispatcherSystem)
+            public override void InvokeOnAuthorityChangeLifecycleCallbacks(Dictionary<int, MonoBehaviourActivationManager> entityIndexToManagers)
+            {
+                var authoritiesChangedTags = AuthoritiesChangedComponentGroup.GetComponentArray<AuthoritiesChanged<<#= componentDetails.TypeName #>>>();
+                var entities = AuthoritiesChangedComponentGroup.GetEntityArray();
+                for (var i = 0; i < entities.Length; i++)
+                {
+                    var activationManager = entityIndexToManagers[entities[i].Index];
+                    for (var j = 0; j < authoritiesChangedTags[i].Buffer.Count; j++)
+                    {
+                        activationManager.ChangeAuthority(<#= unityComponentDefinition.Id #>, authoritiesChangedTags[i].Buffer[j]);
```

Comment:
Preference for GetComponentId(). Eventually we should replace the function with a field though.

Reply:
It's a function because it's used by the methods implemented in the abstract superclass too. Not sure that's actually good though, might be more readable to have all the method bodies in one place and then we could make this a field too

Reply:
I decided to move the 2 methods implemented in the superclass down to the subclasses, moving the method into a const field.